### DISCUSSION TITLE
Crossbar.io website down

### DISCUSSION THREAD

**Discussion by d-curtis** [crossbario/crossbar#2101] [Q&A] [answered]
Hi!

I noted the [Crossbar.io website](http://crossbar.io/) has been offline for some time.
Thankfully, [The Wayback Machine](https://web.archive.org/web/20230715073248/https://crossbar.io/) has snapshots, but are there any plans to bring back the live website?

**Comment by oberstet** [comment]
Hi Dave, the website isn't coming back because the company is gone. contents wise, the website had at least 2 things that could be worth bringing back: Crossbar.io docs and quite a number of technical blog posts. fwiw, in my eyes, at least the docs would probably best be moved, consolidated and cleaned up at ReadTheDocs like https://autobahn.readthedocs.io/. Have a nice day! Cheers, /Tobias

**Reply by d-curtis** [reply]
Hi @oberstet - thanks for the quick response. I hadn't realised that the company had dissolved, couldn't find any news posts anywhere.
What does this mean for the future of Crossbar router / various Autobahn flavours?

**Reply by oberstet** [reply]
First of all, and most importantly for users IMO is: Crossbar.io, all Autobahn flavors, and WAMP itself is open-source or open standards, and will remain so.

WAMP has (as you likely know) has multiple, third party implementations, and work among WAMP implementers is active and continuing. This is mostly about polishing the spec, but also filling in some gaps that still remained. I am also involved in this.

Then, there is multi-node operation (networks of router nodes), which Crossbar.io already implements, and "it works" - but not yet perfectly.

Finally, there one crucial bigger gap: full transparent end-to-end encryption and multi-operator / peer-to-peer / decentralized operation. There is working alpha-level code in Crossbar.io/Autobahn for that.

I am also continuing using Crossbar.io and all that in projects, actually I just finished a consulting/commercial project using it in summer. It's just a good technology and makes sense. And as part of such projects, I contribute code/fixes and such. Also, unfortunately I need to earn money lacking a rich uncle or such;)

Anyways, my personal interest in Crossbar.io is now tied to serving as a OSS reference platform for WAMP, and in adding a first implementation of end-to-end encryption and decentralized WAMP.

So Crossbar.io/Autobahn will remain the "OSS _reference_ platform" for WAMP and I will continue to contribute there.

Other than that, I am moving to "embedded SW" as from an implementation PoV, the one thing that always nagged me is the fact that Crossbar.io requires a Posix run-time and can't go to say a 256k RAM MCU.

Long story short, if I can get to it, there will be a "Crossbar.io Metal" thing, a completely new Rust based implementation, which can then run on anything from a tiny MCU core, a WASM service instance, or a top-end server CPU on Posix. I've already figured out a detailed design and did some first experiments. It'll be quite some work. Crossbar.io+Autobahn is >100k LoC Python. But the core is quite smaller. I am confident it could be done.

Being able to run networks of router nodes spread from cloud to embedded seamlessly, and transparent, and doing so with fully end-to-end encrypted app traffic, and allowing networks of different operators to tie up meshes, that's the end goal.

Anyways, hope this gives some clues, pls don't hesitate to ask more if you like / are interested.

**Reply by readingtype** [reply]
Hi

For the blog posts there is the wayback machine: [https://web.archive.org/web/20230525051208/https://crossbario.com/blog/](https://web.archive.org/web/20230525051208/https://crossbario.com/blog/)

For the docs: I suggested, on another thread, using PyDoc to build a local copy, which is what I've done myself (and it works well in long railway tunnels...)

Looking forward to crossbar.io/metal :-)

Ben

**Comment by oberstet** [comment]
> Looking forward to crossbar.io/metal :-)

it won't be OSS though, so don't wait;)

**Comment by bytebuff** [comment]
I am very excited to see the news about Crossbar.io Metal. Implementing Crossbar using Rust should significantly increase concurrency.

**Comment by vaibhav-rbs** [comment]
Hello [oberstet](https://github.com/oberstet), I am writing to get your advise. At my current company we use crossbar/autobahn for doing communicating with robots spread across the world. Cossbar going down was a setback to us, but now company have decided to go with some other framework/protocol. we are actively considering to use grpc for our pub/sub and rpc communication.  I wanted to know if you know of some projects who might have moved from crossbar.io to grpc framework. I know it might be a longshot to ask this, but I think it's worth the try.

**Reply by bytebuff** [reply, marked as answer]
I think although Crossbar has fallen, this open-source crossbar still exists. We should continue to develop on this basis instead of turning to other frameworks. Have you encountered any difficulties in using Crossbar?

**Reply by vaibhav-rbs** [reply]
Hi, currently I am facing the issue that CPU utilization is shooting up to 100% even when we have only 8-10 system connected to it and sending traffic, because of that we are seeing slowness in getting the data. we are stuck on version == 20.7.1 and moving from there is currently difficult, we were looking for a way by which we can increase the number of cores alleviating slowness.

**Comment by oberstet** [comment]
@bytebuff thank you for that comment! =)

tbh, it is very hard for me to comment on @vaibhav-rbs post.

sure, I can't expect anyone to look at all of the story from my perspective/background, and at this point, I don't even know for sure what that would be anyways. I mean what conclusion I should take from it.

let me try to give you some color: I am living in a world where marketing trumps technical excellence. that's a fact I have to draw from experience. what is also true is: this is not a field where I excel. 

on the one hand, I just love deep tech, cracking nuts - and do tech in the _right_ way.

on the other hand, the world is like it is, and I want to change or adopt rather than unsuccessfully/pointlessly and depressingly trying to change it.

nevertheless, the human world is the result of people's actions. if no one would ever have done "change" we would still be living in the mud I guess.

so for the moment, what really is helping my brain seems watching and listening to stuff like this https://youtu.be/6tmbeLTHC_0?si=IwiwPqLO7z0T746E&t=981 - very obviously much bigger than all of this before;)

**Comment by jacobanana** [comment]
Hi all,

I've build a copy of the docs and hosted them on a github pages: https://jacobanana.github.io/crossbar/
I hope that's helpful!

Adrien
